src/hdmf/utils.py

```diff
@@ -273,6 +274,12 @@ def __parse_args(validator, args, kwargs, enforce_type=True, enforce_shape=True,
                 type_errors.append("missing argument '%s'" % argname)
             else:
                 if enforce_type:
+                    from .container import TermSetWrapper # circular import fix
+                    termset = False
+                    if isinstance(argval, TermSetWrapper):
+                        # kwargs is the dict that stores the object names and the values
+                        # we can use this to unwrap the dataset/attribute to use the "item" for docval to validate the type.
+                        argval = kwargs[argname].item
```

Comment:
Should this be:

```suggestion
                        argval = argval.item
```

I think in: https://github.com/hdmf-dev/hdmf/blob/561e2797e5b488e90640ccc08d978e5c6b8b89b5/src/hdmf/utils.py#L260-L271

``__parse_args`` already extracts the value of the argument and looking up the argument and assigns it to ``argval``. I think looking up the argument via ``kwargs[argname].item`` would fail for positional arguments.

Reply:
I think this check to "unwrap" `argval` may need to go before: 

https://github.com/hdmf-dev/hdmf/blob/561e2797e5b488e90640ccc08d978e5c6b8b89b5/src/hdmf/utils.py#L276

since the other if conditions to check `shape` and `enum` that appear afterwards also need the unwrapped argument value, e.g., here:

https://github.com/hdmf-dev/hdmf/blob/561e2797e5b488e90640ccc08d978e5c6b8b89b5/src/hdmf/utils.py#L290-L291

Reply:
@oruebel it remains unwrapped throughout

Reply:
The order of operations when the code runs is 
1. The wrapper validates the item
2. Then the docval is called
That way we actually just need to unwrap the item once.

Reply:
> @oruebel it remains unwrapped throughout

Yes, but what happens if `enforce_type` is `False` and `enforce_shape` is False? In that case the code to unwrap would not be called and `argval` should still be wrapped.

Reply:
Ahh I see. In practice, why would we ever not enforce type?

Reply:
> Ahh I see. In practice, why would we ever not enforce type?

I'm not sure if/were this might happen right now  But I think all that is needed is to move the few lines of code that do the unwrapping before the `if enforce_type` statement so that the unwrapping is always done before any of the checks.

Reply:
Makes sense and will do. Sorry for that miss.